asf_heat_pump_suitability/pipeline/prepare_features/protected_areas.py

```diff
@@ -6,6 +6,27 @@
 from asf_heat_pump_suitability.getters import get_datasets
 
 
+def load_transform_df_uprn_in_protected_area(gdf: gpd.GeoDataFrame) -> pl.DataFrame:
+    """
+    Generate dataframe of UPRNs located within building conservation areas in England and Wales and UPRNS
+    located within Scottish World Heritage sites.
+
+    Args:
+        gdf (pl.DataFrame): dataframe with point geometries per UPRN in BNG
+
+    Returns:
+        pl.DataFrame: EPC UPRNs in building conservation areas in England and Wales and Scottish World Heritage Sites
+    """
+    ew_df = generate_df_uprn_in_cons_area(gdf).rename(
+        {"in_conservation_area_ew": "in_protected_area"}
+    )
+    s_df = generate_df_uprn_in_whs(gdf).rename(
+        {"in_world_heritage_site": "in_protected_area"}
+    )
+
+    return pl.concat([ew_df, s_df])
```

Comment:
Uh oh, you have found a bug!! Thank you! :D 

This is definitely a problem and needs to be fixed. Here we have ended up with not only duplicate UPRNs, but duplicated UPRNs with opposing `in_protected_areas` values!

This is because when we check if a UPRN is in a World Heritage Site in Scotland, any that aren't in a WHS are flagged as `False` (including those not even in Scotland). So if a property in England or Wales is in a conservation area, it will receive both a `True` and `False` flag!

Reply:
This should be fixed now with the new code in `generate_df_uprn_in_whs` function below. Now, we filter EPC data to Scotland properties only before checking if they are in a WHS. Then we concat only scottish data to england and wales data (which should just be england and wales due to the fact we are joining with conservation area data with an `inner` join). I checked the resulting df, and there are now no duplicated UPRNs in the combined dataset.